Detectors/EMCAL/simulation/include/EMCALSimulation/SimParam.h

```diff
@@ -82,32 +76,51 @@ class SimParam
   void setECPrimaryThreshold(Float_t val) { mECPrimThreshold = val; }
   Float_t getECPrimaryThreshold() const { return mECPrimThreshold; }
 
+  void setSignalDelay(Float_t val) { mSignalDelay = val; }
```

Comment:
The setters are not compatible with the way ConfigurableParamHelper works:
the ``SimParam::Instance()`` returns ``const ref``, so the setters are useless (unless you clone the SimParam). The proper way to set the singletone data members is via static method, e.g.
``o2::conf::ConfigurableParam::updateFromString("EMCSimParam.mA=1")`` or from the ini file with similar entries via ``o2::conf::ConfigurableParam::updateFromFile(inifile)``

@sawenzel , please correct me if I am wrong.

Reply:
That's correct. We can discuss other ways if there is a use-case not currently covered.